wordpress-mobile/WordPress-iOS
# Add local cache for What's New - feature announcements

Fixes #14585

WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/279

To test:

- put two breakpoints at lines 18 and 21 in `AnnouncementsCache.swift`
- go to Me/App Settings/What's New in WordPress
- Make sure that the execution pauses at line 21
- Resume the execution and make sure that the screen looks like the examples below
- do it again (except the breakpoints, that are already there)
- Make sure this time the execution pauses at line 18
- While you're paused, take a loot at the contents of `UserDefaults.announcements` and make sure they look consistent with the screenshots above
- Resume the execution and make sure that the screen has not changed

To reset the `UserDefaults`, pause the debugger and type

`e -l swift -- import WordPress; UserDefaults.standard.announcements = nil`

iPhone | iPad
--- | ---
<img height="400" src="https://user-images.githubusercontent.com/34376330/92183713-cb8d1980-ee14-11ea-962a-7c484a3cfcd4.png"> | <img height="400" src="https://user-images.githubusercontent.com/34376330/92183739-d9db3580-ee14-11ea-8224-39fc4e7f7bc4.png">


PR submission checklist:

- [x] ~~I have considered adding unit tests where possible.~~ Future PR.
- [x] ~~I have considered adding accessibility improvements for my changes.~~ Future PR.
